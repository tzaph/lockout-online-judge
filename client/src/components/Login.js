import React, { useState } from "react";
import { Alert } from "react-bootstrap";
import { useAuth } from "../contexts/AuthContext";
import { Link, useNavigate } from "react-router-dom";

const Login = () => {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const login = useAuth()?.login;
  const [error, setError] = useState("");
  const [loading, setLoading] = useState(false);
  const navigate = useNavigate();

  const handleSubmit = async (e) => {
    e.preventDefault();
    try {
      await login(email, password);
      setLoading(true);
      setError("");
      navigate("/");
    } catch {
      setError("Failed to login");
    }
    setLoading(false);
  };

  return (
    <div className="auth-form-container">
      <h2>Login</h2>
      {error && <Alert variant="danger">{error}</Alert>}
      <form className="register-form" onSubmit={handleSubmit}>
        <label for="email">Email</label>
        <input
          value={email}
          onChange={(e) => setEmail(e.target.value)}
          type="email"
          placeholder="youremail@gmail.com"
          id="email"
          name="email"
        />
        <label for="password">Password</label>
        <input
          value={password}
          onChange={(e) => setPassword(e.target.value)}
          type="password"
          placeholder="********"
          id="password"
          name="password"
        />
        <button disabled={loading} type="submit">
          Login
        </button>
      </form>
      <div className="link-button">
        Don't have an account? <Link to="/register">Register</Link>
      </div>
    </div>
  );
};

export default Login;
