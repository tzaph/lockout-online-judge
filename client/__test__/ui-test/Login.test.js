import React from "react";
import renderer from "react-test-renderer";
import Login from "../../src/components/Login";
import { BrowserRouter as Router } from "react-router-dom";

describe("<Login />", () => {
  it("Renders correctly", () => {
    const tree = renderer
      .create(
        <Router>
          <Login />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
