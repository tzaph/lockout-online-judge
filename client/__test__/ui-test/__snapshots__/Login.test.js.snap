// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> Renders correctly 1`] = `
<div
  className="auth-form-container"
>
  <h2>
    Login
  </h2>
  <form
    className="register-form"
    onSubmit={[Function]}
  >
    <label
      for="email"
    >
      Email
    </label>
    <input
      id="email"
      name="email"
      onChange={[Function]}
      placeholder="youremail@gmail.com"
      type="email"
      value=""
    />
    <label
      for="password"
    >
      Password
    </label>
    <input
      id="password"
      name="password"
      onChange={[Function]}
      placeholder="********"
      type="password"
      value=""
    />
    <button
      disabled={false}
      type="submit"
    >
      Login
    </button>
  </form>
  <div
    className="link-button"
  >
    Don't have an account? 
    <a
      href="/register"
      onClick={[Function]}
    >
      Register
    </a>
  </div>
</div>
`;
